### PR TITLE
Fallback icon

### PR DESCRIPTION
**Pull request by rafa-faj**
- Added fallback icon when no icon is available for an amenity

### CODE REVIEW

File: FrontEnd/src/components/basics/Amenities/Amenities.stories.tsx [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
 export default {
   title: 'Amenities',
   component: Amenities,
 };
 
-const Template = StoryTemplate<
-  ComponentProps<typeof Amenities>,
-  typeof Amenities
->(Amenities);
+const Template: Story<ComponentProps<typeof Amenities>> = (args) => (
```
Comment (bamit5) [inline FrontEnd/src/components/basics/Amenities/Amenities.stories.tsx:10]: Why not use the StoryTemplate?

File: FrontEnd/src/utils/storybook/storybook.ts [hunk 2]
```diff
@@ -1 +0,0 @@
-export const isRunningInStorybook = () => process.env.STORYBOOK === 'true';
```
Comment (bamit5) [inline FrontEnd/src/utils/storybook/storybook.ts:1]: Why was this removed? I think it's used in the Link component

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,23 @@
+LANG = python3
```
Comment (bamit5) [inline Makefile:1]: You shouldn't make backend changes in a frontend PR. Confirm with Cris that this file is good to go, in the future try to separate backend and frontend changes into their own PRs please

File: FrontEnd/src/components/basics/Fallback/Fallback.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import React, { FunctionComponent } from 'react';
+import { amenityToIcon, Amenity } from '../Amenities/Amenities';
```
Comment (bamit5) [inline FrontEnd/src/components/basics/Fallback/Fallback.tsx:2]: prefer `from '@basics'`

File: FrontEnd/src/components/basics/Amenities/Amenities.tsx [hunk 5]
```diff
@@ -6,6 +6,7 @@ import React, {
 import { amenityIcons, IconProps } from '@icons';
 import Col, { ColProps } from 'react-bootstrap/Col';
 import styles from './Amenities.module.scss';
+import FallbackIcon from '../Fallback/Fallback';
```
Comment (bamit5) [inline FrontEnd/src/components/basics/Amenities/Amenities.tsx:9]: `from '@basics'`

File: FrontEnd/src/components/basics/Amenities/Amenities.tsx [hunk 6]
```diff
@@ -102,8 +103,9 @@ const Amenities: FunctionComponent<AmenitiesProps> = ({
   return (
     <div className={cn(styles.wrapperDefault, className)}>
       {selected.map((s) => {
-        const SelectedIcon = amenityToIcon[s];
-        const Icon = () => (variant !== 'onlyLabel' ? <SelectedIcon /> : null);
+        // const SelectedIcon = amenityToIcon[s]; // change to fallback
```
Comment (bamit5) [inline FrontEnd/src/components/basics/Amenities/Amenities.tsx:106]: remove commented out code

File: FrontEnd/src/components/basics/Fallback/Fallback.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import React, { ComponentProps } from 'react';
+import { Story } from '@storybook/react';
+import FallbackIcon from './Fallback';
+
+export default {
+  title: 'FallbackIcon',
+  component: FallbackIcon,
+};
+
+const Template: Story<ComponentProps<typeof FallbackIcon>> = (args) => (
```
Comment (bamit5) [inline FrontEnd/src/components/basics/Fallback/Fallback.stories.tsx:10]: Why not use StoryTemplate

File: FrontEnd/src/components/basics/Fallback/Fallback.tsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+import { Icon as IconType, miscIcons } from '@icons';
+
+// Amenity needs to be any so it can be type guarded
+interface FallbackProps {
+  amenity: any;
```
Comment (bamit5) [inline FrontEnd/src/components/basics/Fallback/Fallback.tsx:7]: If I understand correctly, FallbackIcon is meant to be a Fallback _only_ for the following case: when something that was meant to be an icon (ie received an icon key from the backend) is actually not in the list of icons on the frontend.

So in this case, we should only pass in the type of `Amenity`. This FallbackIcon _expects_ an Amenity, but will provide a _Fallback_ in case the string passed through is actually not an Amenity. Does that make sense? Reach out if not and I can explain further

File: FrontEnd/src/components/basics/Fallback/Fallback.tsx [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+// Type guard to check if key is of Amenity type
```
Comment (bamit5) [inline FrontEnd/src/components/basics/Fallback/Fallback.tsx:10]: nit: the best syntax for header comments for functions/classes is:
`[name here]` is ....
so in this case: `isAmenity is a type guard to...`

File: FrontEnd/src/components/basics/Fallback/Fallback.tsx [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+const isAmenity = (key: any): key is Amenity => {
+  return key in amenityToIcon;
+};
+
+// Fallback Icon Component (could be a normal amenity svg too)
```
Comment (bamit5) [inline FrontEnd/src/components/basics/Fallback/Fallback.tsx:15]: what does `normal amenity svg` mean? And I think a more descriptive comment of the purpose of FallbackIcon would be helpful. We want as much written down as possible so that in the future, we don't forget and don't have to ask co-workers 'what does this do? why is it useful?'

File: FrontEnd/src/components/basics/Fallback/Fallback.tsx [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+const FallbackIcon: FunctionComponent<FallbackProps> = ({ amenity }) => {
+  const Icon: IconType = isAmenity(amenity)
```
Comment (bamit5) [inline FrontEnd/src/components/basics/Fallback/Fallback.tsx:17]: Can shorten the function to the following:
```
const FallbackIcon: FunctionComponent<FallbackProps> = ({ amenity }): IconType => (
  isAmenity(amenity)
    ? amenityToIcon[amenity]
    : miscIcons.Fallback
);
```

**Review by bamit5 — CHANGES_REQUESTED** [review]

File: FrontEnd/src/components/basics/Fallback/Fallback.tsx (hunk 11 above)
Comment (rafa-faj) [reply FrontEnd/src/components/basics/Fallback/Fallback.tsx:17]: Storybook wasn't liking this change for some reason, so I kept it how it was originally

File: FrontEnd/src/components/basics/Fallback/Fallback.tsx [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+// isAmenity is a type guard to check if key is of Amenity type
+const isAmenity = (key: any): key is Amenity => {
+  return key in amenityToIcon;
+};
+
+// Fallback Icon Component returns an amenity Icon if it is in the list of amenity icons
+// otherwise, returns a fallback Icon
+const FallbackIcon: FunctionComponent<FallbackProps> = ({ amenity }) => {
```
Comment (bamit5) [inline FrontEnd/src/components/basics/Fallback/Fallback.tsx:16]: Nit: change to `AmenityIcon` and move component to under `Amenities` folder since this'll mostly be used in the Amenities component (also change to `AmenityIcon.tsx` or something similar rather than `Fallback.tsx`)

File: FrontEnd/src/components/basics/Amenities/Amenities.tsx [hunk 13]
```diff
@@ -102,8 +103,8 @@ const Amenities: FunctionComponent<AmenitiesProps> = ({
   return (
     <div className={cn(styles.wrapperDefault, className)}>
       {selected.map((s) => {
-        const SelectedIcon = amenityToIcon[s];
-        const Icon = () => (variant !== 'onlyLabel' ? <SelectedIcon /> : null);
+        const SelectedIcon = <FallbackIcon amenity={s} />;
```
Comment (bamit5) [inline FrontEnd/src/components/basics/Amenities/Amenities.tsx:106]: nit: change to `selectedIcon` (capitalize when it's used as a Component but this isn't)

**Review by bamit5 — CHANGES_REQUESTED** [review]
